bintree.go

```diff
@@ -29,13 +29,36 @@ func binTreeNodeTypeToString(nodeType BinTreeNodeType) string {
 	return "??ERROR??"
 }
 
-type binTreeNode struct {
```

Comment:
The key here is to separate the parent/left/right from the actual binTreeNode, so SimpleParser can utilize it instead of having its own TreeNode structure.